src/plugins/action_update_review_assignments/src/databaseMethods.ts

```diff
@@ -0,0 +1,68 @@
+const databaseMethods = (DBConnect: any) => ({
+  getReviewAssignmentById: async (reviewAssignmentId: number) => {
+    const text = `
+        SELECT * FROM review_assignment
+        WHERE id = $1
+        `
+    try {
+      const result = await DBConnect.query({
+        text,
+        values: [reviewAssignmentId],
+      })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getMatchingReviewAssignments: async (
+    reviewAssignmentId: number,
+    applicationId: number,
+    stageNumber: number,
+    reviewLevel: number
+  ) => {
+    const text = `
+    SELECT * FROM review_assignment
+    WHERE application_id = $2
+    AND stage_number = $3
+    AND level = $4
+    AND id <> $1
+    `
+    try {
+      const result = await DBConnect.query({
+        text,
+        values: [reviewAssignmentId, applicationId, stageNumber, reviewLevel],
+      })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  updateReviewAssignments: async (reviewAssignments: any) => {
+    const reviewAssignmentUpdateResults = []
+    for (const reviewAssignment of reviewAssignments) {
+      const { id, status } = reviewAssignment
+      const text = `
+      UPDATE review_assignment
+      SET status = $2
+      WHERE id = $1
```

Comment:
maybe 'any' once again ?

Reply:
Actually, doesn't quite work as easily, since `reviewAssignments` is an array of objects with {id, status}, and each of the records to be updated gets a (possibly) different status, so we need to loop over them one by one.